oort/oort-azure/src/main/groovy/com/netflix/spinnaker/oort/azure/model/AzureResourceRetriever.groovy

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright 2015 The original authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.oort.azure.model
+
+import com.microsoft.azure.management.network.models.LoadBalancer
+import com.netflix.spinnaker.clouddriver.azure.security.AzureCredentials
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider
+import com.netflix.spinnaker.clouddriver.azure.config.AzureConfigurationProperties
+import com.netflix.spinnaker.clouddriver.azure.security.AzureNamedAccountCredentials
+import com.netflix.spinnaker.clouddriver.azure.client.models.AzureLoadBalancerDescription
+import com.netflix.spinnaker.oort.model.HealthState
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+
+import javax.annotation.PostConstruct
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.locks.Lock
+import java.util.concurrent.locks.ReentrantLock
+
+@Slf4j
+class AzureResourceRetriever {
```

Comment:
I reckon this is one area that Google didn't follow the AWS conventions.

Specifically we should be building caching agents corresponding to each of the data types ... see the CachingAgent subclasses.

The issue with running everything in memory is two fold ... (a) not everything will necessarily fit and (b) you're really locked to having a single instance of clouddriver otherwise you run the risk of cache inconsistencies.

I believe @duftler has this on the list of things to revisit.